griptape/drivers/prompt/openai_chat_prompt_driver.py

```diff
@@ -7,58 +7,106 @@
 from griptape.drivers import BasePromptDriver
 from griptape.tokenizers import OpenAiTokenizer
 from typing import Tuple, Type
+from pytimeparse.timeparse import timeparse
+from datetime import datetime, timedelta
+import requests
 
 
 @define
 class OpenAiChatPromptDriver(BasePromptDriver):
     """
     Attributes:
         api_type: Can be changed to use OpenAI models on Azure.
-        api_version: API version. 
+        api_version: API version.
         api_base: API URL.
         api_key: API key to pass directly; by default uses `OPENAI_API_KEY_PATH` environment variable.
         max_tokens: Optional maximum return tokens. If not specified, no value will be passed to the API. If set, the
             value will be bounded to the maximum possible as determined by the tokenizer.
         model: OpenAI model name. Uses `gpt-4` by default.
         organization: OpenAI organization.
         tokenizer: Custom `OpenAiTokenizer`.
-        user: OpenAI user. 	
+        user: OpenAI user.
     """
+
     api_type: str = field(default=openai.api_type, kw_only=True)
     api_version: Optional[str] = field(default=openai.api_version, kw_only=True)
     api_base: str = field(default=openai.api_base, kw_only=True)
     api_key: Optional[str] = field(default=Factory(lambda: os.environ.get("OPENAI_API_KEY")), kw_only=True)
     organization: Optional[str] = field(default=openai.organization, kw_only=True)
     model: str = field(kw_only=True)
     tokenizer: OpenAiTokenizer = field(
-        default=Factory(lambda self: OpenAiTokenizer(model=self.model), takes_self=True),
-        kw_only=True
+        default=Factory(lambda self: OpenAiTokenizer(model=self.model), takes_self=True), kw_only=True
     )
     user: str = field(default="", kw_only=True)
-    ignored_exception_types: Tuple[Type[Exception], ...] = field(default=Factory(lambda: (openai.InvalidRequestError)), kw_only=True)
+    ignored_exception_types: Tuple[Type[Exception], ...] = field(
+        default=Factory(lambda: openai.InvalidRequestError), kw_only=True
+    )
+    __ratelimit_request_limit: Optional[int] = field(default=None)
+    __ratelimit_requests_remaining: Optional[int] = field(default=None)
+    __ratelimit_requests_reset_at: Optional[datetime] = field(default=None)
+    __ratelimit_token_limit: Optional[int] = field(default=None)
+    __ratelimit_tokens_remaining: Optional[int] = field(default=None)
+    __ratelimit_tokens_reset_at: Optional[datetime] = field(default=None)
```

Comment:
I think we can accomplish this private pattern by using `init=False` as described [here](https://www.attrs.org/en/stable/examples.html#basics).

i.e. `_ratelimit_request_limit: Optional[int] = field(init=False, default=None)`